README.md

```diff
@@ -143,6 +143,49 @@ bottomBar.setOnTabReselectListener(new OnTabReselectListener() {
 });
 ```
 
+### Intercepting tab selections
+
+If you want to conditionally cancel selection of any tab, you absolutely can. Just assign a ```TabSelectionInterceptor``` to the BottomBar, and return true from the ```shouldInterceptTabSelection()``` method.
+
+```java
+bottomBar.setTabSelectionInterceptor(new TabSelectionInterceptor() {
+    @Override
+    public boolean shouldInterceptTabSelection(@IdRes int oldTabId, @IdRes int newTabId) {
+        if (newTabId == R.id.tab_pro_feature && !userHasProVersion()) {
+          startProVersionPurchaseFlow();
+          return true;
+        }
+        
+        return false;
+    }
+});
+```
+
+### Changing icons based on selection state
+
+If you want to have different icon when a specific tab is selected, just use state list drawables.
+
+**res/drawable/my_tab_icon.xml**
+
+```xml
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/ic_myicon_selected" android:state_selected="true" />
+    <item android:drawable="@drawable/ic_myicon_default" android:state_selected="false" />
```

Comment:
<item android:drawable="@drawable/ic_myicon_default" />

The false if not needed because the last thing is always the default